### PR TITLE
new resource: azurerm_log_analytics_workspace_datasource_linux_syslog

### PR DESCRIPTION
**Pull request by magodo**
This implements one of the log analytics workspace data sources required in terraform-providers/terraform-provider-azurerm#3182

NOTE: There is a dedicated workspace-level resource called `linux_syslog_collection`, which has only a `state` indicating whether `linux_syslog` DS is applied to all the machines in this workspace. Rather than creating a corresponding resource for the "collection", making it a workspace property seems more intuitive in the context of terraform.

## Test result

```bash
💤 via 🦉 v1.14.1 make testacc TEST=./azurerm/internal/services/loganalytics/tests TESTARGS="-run='(TestAccAzureRMLogAnalyticsWorkspace_update|TestAccDataSourceAzureRMLogAnalyticsWorkspace_basic)'"
==> Checking that code complies with gofmt requirements...
==> Checking that Custom Timeouts are used...
TF_ACC=1 go test ./azurerm/internal/services/loganalytics/tests -v -run='(TestAccAzureRMLogAnalyticsWorkspace_update|TestAccDataSourceAzureRMLogAnalyticsWorkspace_basic)' -timeout 180m -ldflags="-X=github.com/terraform-providers/terraform-provider-azurerm/version.ProviderVersion=acc"
=== RUN   TestAccDataSourceAzureRMLogAnalyticsWorkspace_basic
=== PAUSE TestAccDataSourceAzureRMLogAnalyticsWorkspace_basic
=== RUN   TestAccAzureRMLogAnalyticsWorkspace_update
=== PAUSE TestAccAzureRMLogAnalyticsWorkspace_update
=== CONT  TestAccDataSourceAzureRMLogAnalyticsWorkspace_basic
=== CONT  TestAccAzureRMLogAnalyticsWorkspace_update
--- PASS: TestAccDataSourceAzureRMLogAnalyticsWorkspace_basic (220.81s)
--- PASS: TestAccAzureRMLogAnalyticsWorkspace_update (405.41s)
PASS
ok      github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/loganalytics/tests  405.450s
```

### CODE REVIEW

File: azurerm/internal/services/loganalytics/resource_arm_log_analytics_datasource_linux_syslog.go [hunk 1]
```diff
@@ -0,0 +1,236 @@
...
+}
+
+type dataSourceLinuxSyslog struct {
+	Name       string                          `json:"syslogName"`
+	Severities []dataSourceLinuxSyslogSeverity `json:"syslogSeverities"`
+}
+
+type dataSourceLinuxSyslogSeverity struct {
+	Severity string `json:"severity"`
+}
```
Comment (katbyte) [inline azurerm/internal/services/loganalytics/resource_arm_log_analytics_datasource_linux_syslog.go:115]: Shouldn't this be added to the sagger and become part of the SDK?

File: azurerm/internal/services/loganalytics/resource_arm_log_analytics_datasource_linux_syslog.go [hunk 2]
```diff
@@ -0,0 +1,236 @@
...
+		},
+	}
+
+	if _, err := client.CreateOrUpdate(ctx, resourceGroup, workspaceName, name, param); err != nil {
+		return fmt.Errorf("failed to create Log Analytics DataSource Linux Syslog %q (Resource Group %q / Workspace: %q): %+v", name, resourceGroup, workspaceName, err)
+	}
+
+	resp, err := client.Get(ctx, resourceGroup, workspaceName, name)
+	if err != nil {
+		return fmt.Errorf("failed to retrieve Log Analytics Data Source Linux Syslog %q (Resource Group %q / Workspace: %q): %+v", name, resourceGroup, workspaceName, err)
```
Comment (katbyte) [inline azurerm/internal/services/loganalytics/resource_arm_log_analytics_datasource_linux_syslog.go:153]: replacing error with failed doesn't really change the fact we start the error message with a ERROR or FAILED. maybe we should change to grammar like
```suggestion
		return fmt.Errorf("retrieving Log Analytics Data Source Linux Syslog %q (Resource Group %q / Workspace: %q): %+v", name, resourceGroup, workspaceName, err)
```

File: azurerm/internal/services/loganalytics/resource_arm_log_analytics_datasource_linux_syslog.go [hunk 3]
```diff
@@ -0,0 +1,236 @@
...
+	}
+	if resp.ID == nil || *resp.ID == "" {
+		return fmt.Errorf("cannot read Log Analytics Data Source Linux Syslog %q (Resource Group %q / Workspace: %q) ID", name, resourceGroup, workspaceName)
```
Comment (katbyte) [inline azurerm/internal/services/loganalytics/resource_arm_log_analytics_datasource_linux_syslog.go:156]: and
```suggestion
		return fmt.Errorf("reading Log Analytics Data Source Linux Syslog %q (Resource Group %q / Workspace: %q) ID", name, resourceGroup, workspaceName)
```

as the error or failed is implied? YDYT?

**Comment by magodo** [comment]
@katbyte Thank you for the quick response 👍
I have made the update according to your comment. For the type issue, I have submitted a PR (Azure/azure-rest-api-specs#9072) to azure-api-spec to track and once it's been addressed, then we can switch to the SDK type.

**Comment by magodo** [comment]
@katbyte is it possible to get this merged for now (at least it works)?
Per [reply](https://github.com/Azure/azure-rest-api-specs/issues/9072#issuecomment-614625155) from service team, the issue seems not likely to be addressed in the near future.

**Comment by tombuildsstuff** [comment]
hey @magodo 

Unfortunately we need the SDK to define these types else they can change on Azure's side without correctly handling semantic versioning - as such these types need to be present in the Azure SDK/Swagger here. Since this is currently blocked [on this upstream API Issue](https://github.com/Azure/azure-rest-api-specs/issues/9072) - I'm going to temporarily close this issue for the moment (it's already in the Blocked milestone) - but once the upstream issue has been resolved then we can re-open this and take another look.

Thanks!

**Pull request closed without merging**